### ISSUE TITLE
Upgrade to Kubernetes 1.6.2 fleet cannot start correctly

### ISSUE THREAD

**Issue by kkdai** [TheNewNormal/kube-cluster-osx#114]
After "Change Kubernetes version" to 1.6.2
Restart Kube-cluster will occur following error:

```
Starting Kubernetes fleet units ...
####################################################################
WARNING: fleetctl (0.11.7) is older than the latest registered
version of fleet found in the cluster (0.11.8). You are strongly
recommended to upgrade fleetctl to prevent incompatibility issues.
####################################################################
```
Any suggestion?

**Comment by rimusz** [comment]
k8s v1.6.2 is not supported yet, too many changes needed, sorry I'm a bit short of time for it.

**Comment by ncheneweth** [comment]
what if I wanted to fork this repo and attempt rebuilding with later versions? Is there any written documentation on building?

**Comment by rimusz** [comment]
@ncheneweth it is Xcode project, it is easy to build with it.
The changes for k8s v1.6/1.7 support are needed in bash scripts `*.command` and `functions.sh`.
Also  k8s v1.6/1.7 needs etcd v3, the change has to go to https://github.com/TheNewNormal/kube-cluster-osx/blob/master/src/cloud-init/user-data.master and https://github.com/TheNewNormal/kube-cluster-osx/blob/master/src/cloud-init/user-data.node

**Comment by ncheneweth** [comment]
given the deprecation of fleet - had you in mind to refactor that implementation out in the next version?

**Comment by rimusz** [comment]
yes, I have plans for refactoring, but is not a small job :) just need to find a spare time for it.
Think to use bootkube or kubeadm for it.
